src/utils/utils.cairo

```diff
@@ -547,6 +547,320 @@ namespace Helpers {
         return ();
     }
 
+    // @notice Splits a felt into 16 bytes, big-endian, and outputs to `dst`.
+    func bytes_used_128{range_check_ptr}(value: felt) -> felt {
+        // Fill dst using only hints with no opcodes.
+        let offset = 0;
+        tempvar base = 256 ** 15;
+        let bound = base;
+        tempvar max = base - 1;
+        let (dst) = alloc();
+
+        // 0.
+        let output = &dst[offset];
+        let offset = offset + 1;
+        %{
+            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
+            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
+        %}
+        tempvar x = [output];
+        tempvar is_x_lt_value = is_le(x + 1, value);
+        if (is_x_lt_value != FALSE) {
+            return 16 - offset + 1;
+        }
+
+        [range_check_ptr] = value;
```

Comment:
you need to make the range check before any other things otherwise the prover may cheat

Reply:
you need to check that `[output]` is in range, not the input (`value`)